torch/nn/attention/flex_attention.py

```diff
@@ -840,7 +840,7 @@ def _apply_kernel_options(query, key, value, kernel_options):
     output_logsumexp = any_inputs_require_grad and torch.is_grad_enabled()
     kernel_options["OUTPUT_LOGSUMEXP"] = output_logsumexp
 
-    if query.size(-2) >= 128 and (query.size(-2) % 128 != 0 or key.size(-2) % 128 != 0):
```

Comment:
@drisspg I remembered we all agreed to move this kernel options inside of lowering, but I'd like to revisit this decision. Since this is an option across all kernels(attn fwd, attn bwd, decoding) and it's decided only by the input shape, so it makes more sense to keep it here? Otherwise, we need three copies.

Reply:
Chatted offline, we aligned to move it to lowering function.